mcda/utils/utils_for_main.py

```diff
@@ -322,10 +361,15 @@ def read_matrix(input_matrix_path: str) -> pd.DataFrame:
     :param input_matrix_path: str
     :rtype: pd.DataFrame
     """
+    try:
+        ensure_directory_exists(input_directory_path)
+    except IOError:
+        logging.error(f"Reading a matrix failed.")
 
     try:
-        filename = input_matrix_path
-        with open(filename, 'r') as fp:
+        full_file_path = os.path.join(input_directory_path, input_matrix_path)
+        with open(full_file_path, 'r') as fp:
```

Comment:
If you don't exit the code above where you are creating input directories for the user who did not give a valid input folder then I believe that here you will get an exception.